six/modules/c++/six.sicd/include/six/sicd/AreaPlaneUtility.h

```diff
@@ -0,0 +1,92 @@
+/* =========================================================================
+ * This file is part of six.sicd-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * six.sicd-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef __SIX_SICD_AREA_PLANE_UTILITY_H__
+#define __SIX_SICD_AREA_PLANE_UTILITY_H__
+
+#include <vector>
+
+#include <six/Types.h>
+#include <six/sicd/ComplexData.h>
+#include <six/sicd/RadarCollection.h>
+
+#include <types/RgAz.h>
+#include <types/RowCol.h>
+
+namespace six
+{
+namespace sicd
+{
+/*!
+ * \class AreaPlaneUtility
+ * \brief Provides functions to populate
+ * ComplexData->RadarCollection->Area->Plane if unset
+ */
+class AreaPlaneUtility
+{
+public:
+    /*!
+     * If data has no areaPlane, derive it and set it
+     * \param data ComplexData to populate with areaPlane
+     * \param includeSegmentList Should the areaPlane's segmentlist be filled?
+     */
+    static void setAreaPlane(ComplexData& data, bool includeSegmentList=true,
+            double sampleDensity=SAMPLE_DENSITY);
```

Comment:
Might as well make `SAMPLE_DENSITY` public in case someone wants to programmatically know what it is for some reason.  Also, I'd rename to `DEFAULT_SAMPLE_DENSITY` to make it more clear.